server/src/main/java/org/opensearch/common/io/stream/NamedWriteableRegistryResponse.java

```diff
@@ -22,28 +22,29 @@
  */
 public class NamedWriteableRegistryResponse extends TransportResponse {
 
-    private final Map<String, Class> registry;
+    private final Map<String, Class<? extends NamedWriteable>> registry;
 
     /**
      * @param registry Map of writeable names and their associated category class
      */
-    public NamedWriteableRegistryResponse(Map<String, Class> registry) {
+    public NamedWriteableRegistryResponse(Map<String, Class<? extends NamedWriteable>> registry) {
         this.registry = new HashMap<>(registry);
     }
 
     /**
      * @param in StreamInput from which map entries of writeable names and their associated category classes are read from
      * @throws IllegalArgumentException if the fully qualified class name is invalid and the class object cannot be generated at runtime
      */
+    @SuppressWarnings("unchecked")
```

Comment:
Generally we try to put `@SuppressWarnings` at the lowest scope possible.  In the constructor we were assigning a value to an instance variable, so we had to use the method level, but here we are assigning to a local variable.  You should be able to put this suppression down above line 47.

Reply:
got it, I will move `@SuppressWarnings` in next commit